### PR TITLE
[shopsys] Start Building Your Application article

### PR DESCRIPTION
**Pull request by simara-svatopluk**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Implementator didn't know what to do after installation. Now we have a starting point explaining the most important steps and concepts
|New feature| No 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: docs/introduction/start-building-your-application.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+Routing is a mechanism that maps URL path to controllers and methods.
+
+*We use Symfony routing, so please find more in the [official documentation](https://symfony.com/doc/3.4/routing.html)*
+
+You can adjust the routing in configuration `src/Shopsys/ShopBundle/Resources/config/routing_front.yml` and [locale-specific](/docs/introduction/how-to-set-up-domains-and-locales.md#32-frontend-routes) in configuration `src/Shopsys/ShopBundle/Resources/config/routing_front_xx.yml`.
+
+## What is a default currency and how to set it up
+
+A default currency is a currency that is displayed when showing a price and no currency is specified explicitly.
```
Comment (TomasLudvik) [inline docs/introduction/start-building-your-application.md]: This sentence is confusing, I do not know what it says.

File: docs/introduction/start-building-your-application.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+The default currency is different for administration and for each of domains and you can adjust the default currency in each place individually.
```
Comment (TomasLudvik) [inline docs/introduction/start-building-your-application.md]:
```suggestion
The default currency is different for administration and for each of domains and you can adjust the default currency for each one individually.
```

File: docs/introduction/start-building-your-application.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+The administration default currency is used in twig templates eg. as `{{ value|priceWithCurrencyAdmin }}`.
+The default currency for domain is used eg. as `{{ cartItemDiscount.priceWithVat|price }}`.
+
+*Please read more in a dedicated article about [price filters](/docs/model/how-to-work-with-money.md#price) and [administration price filter](/docs/model/how-to-work-with-money.md#pricewithcurrencyadmin).*
```
Comment (TomasLudvik) [inline docs/introduction/start-building-your-application.md]:
```suggestion
*Note: Read more in a dedicated article about [price filters](/docs/model/how-to-work-with-money.md#price) and [administration price filter](/docs/model/how-to-work-with-money.md#pricewithcurrencyadmin).*
```

File: docs/introduction/start-building-your-application.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+When you install the project, default currencies are
```
Comment (TomasLudvik) [inline docs/introduction/start-building-your-application.md]:
```suggestion
When you install new project, default currencies are set like this
```

File: docs/introduction/start-building-your-application.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+* `shopsys` *(1st domain)*: `CZK`
+* `2.shopsys` *(2nd domain)*: `EUR`
+* administration: `CZK`
+
+You can change default currencies in administration `Pricing > Currencies`, but this change won't last after application rebuild (operation that you do often during development).
```
Comment (TomasLudvik) [inline docs/introduction/start-building-your-application.md]:
```suggestion
You can change default currencies in administration `Pricing > Currencies`, but this change will not last after application rebuild (operation that you do often during development).
```

File: docs/introduction/start-building-your-application.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+### Adjust demo data
+
+You can adjust the demo data to match your project.
+This takes a bit more effort but once you adjust demo data, the change will be applied after application rebuild.
```
Comment (TomasLudvik) [inline docs/introduction/start-building-your-application.md]:
```suggestion
This takes a bit more effort but once you adjust demo data, the change will be applied every time application is rebuilded.
```

File: docs/introduction/start-building-your-application.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+*Please read more in a dedicated article about [price filters](/docs/model/how-to-work-with-money.md#price) and [administration price filter](/docs/model/how-to-work-with-money.md#pricewithcurrencyadmin).*
+
+When you install the project, default currencies are
+* `shopsys` *(1st domain)*: `CZK`
+* `2.shopsys` *(2nd domain)*: `EUR`
+* administration: `CZK`
+
+You can change default currencies in administration `Pricing > Currencies`, but this change won't last after application rebuild (operation that you do often during development).
+
+### Adjust demo data
```
Comment (TomasLudvik) [inline docs/introduction/start-building-your-application.md]:
```suggestion
### How to set default currency permanently?
```

File: docs/introduction/start-building-your-application.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+You can adjust the demo data to match your project.
+This takes a bit more effort but once you adjust demo data, the change will be applied after application rebuild.
+
+#### Set administration default currency
```
Comment (TomasLudvik) [inline docs/introduction/start-building-your-application.md]:
```suggestion
#### How to set administration default currency
```

File: docs/introduction/start-building-your-application.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+Class `SettingValueDataFixture`, method `load()`
+```diff
++ $eurCurrency = $this->getReference(CurrencyDataFixture::CURRENCY_EUR);
++ $this->setting->set(PricingSetting::DEFAULT_CURRENCY, $eurCurrency->getId());
+```
+
+#### Set first domain default currency
```
Comment (TomasLudvik) [inline docs/introduction/start-building-your-application.md]:
```suggestion
#### How to set first domain default currency
```

File: docs/introduction/start-building-your-application.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+Class `SettingValueDataFixture`, method `load()`
+```diff
++ $eurCurrency = $this->getReference(CurrencyDataFixture::CURRENCY_EUR);
++ $this->setting->setForDomain(PricingSetting::DEFAULT_DOMAIN_CURRENCY, $eurCurrency->getId(), Domain::FIRST_DOMAIN_ID);
+```
+
+#### Set next domains default currency
```
Comment (TomasLudvik) [inline docs/introduction/start-building-your-application.md]:
```suggestion
#### How to set next domains default currency
```

**Comment by simara-svatopluk** [comment]
@TomasLudvik So the structure of the article and the general approach is correct?

**Comment by TomasLudvik** [comment]
> @TomasLudvik So the structure of the article and the general approach is correct?

Good question, I was just thinking about that. It would be probably better if you changed titles and a little descriptions in order to set user the right way. 

It might be something like this:

1. Commit your lock files
2. Set your domains
3. Set your locales ...

With short description why we suggest user to do that.

**Comment by simara-svatopluk** [comment]
Sorry, I don't understand what do you suggest. So wheat do you suggest?

If you suggest numbered headlines, then I have an answer right away:
This article isn't cookbook, it is a list of useful tips that might help. Not steps implementator must do. That's why there aren't numbers in headlines.

File: docs/introduction/start-building-your-application.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
+# Start Building Your Application
+
+Installation of the Shopsys Framework is complete and now you can start building your application.
```
[the remark this answers is not in the document]
Comment (simara-svatopluk) [reply docs/introduction/start-building-your-application.md]: I disagree, the article must be there. Why do you suggest such removal?

File: docs/introduction/start-building-your-application.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+
+```bash
+git add composer.lock package-lock.json migrations-lock.yml
+git commit -m'locked composer, npm and migrations'
+```
+
+## What is a domain and how to set up domains
+
+A domain can be understood as one instance of the shop.
+For example, just furniture can be bough on the domain shopsys-furniture.com while only electronics can be found on the domain shopsys-electro.com.
```
[the remark this answers is not in the document]
Comment (simara-svatopluk) [reply docs/introduction/start-building-your-application.md]: Yes, but it's not purpose of domains, domains purpose is to have different content on different domains even in the same language. Futuremore I almost copied this sentence from existing documentation.

File: docs/introduction/start-building-your-application.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+
+*Please find more in an article that explains the [domain concept fully](/docs/introduction/domain-multidomain-multilanguage.md#domain).*
+
+When you install the project, domains are
+* `shopsys` on the URL `http://127.0.0.1:8000`
+* `2.shopsys` on the URL `http://127.0.0.2:8000`
+
+Please read [how to change domain settings](/docs/introduction/how-to-set-up-domains-and-locales.md#settings-and-working-with-domains) if you have project with only [one domain](/docs/introduction/how-to-set-up-domains-and-locales.md#1-how-to-create-a-single-domain-application) or [more domains](/docs/introduction/how-to-set-up-domains-and-locales.md#2-how-to-add-a-new-domain).
+Make sure you change the [url settings](/docs/introduction/how-to-set-up-domains-and-locales.md#4-change-the-url-address-for-an-existing-domain) at least in the production environment.
```
[the remark this answers is not in the document]
Comment (simara-svatopluk) [reply docs/introduction/start-building-your-application.md]: Changing URL isn't only for "have already running application"... You have to change URLs even for a new project

File: docs/introduction/start-building-your-application.md (hunk 1 above)
Comment (simara-svatopluk) [reply docs/introduction/start-building-your-application.md]: better?

File: docs/introduction/start-building-your-application.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply docs/introduction/start-building-your-application.md]: Describe some situation please

**Comment by TomasLudvik** [comment]
> Sorry, I don't understand what do you suggest. So wheat do you suggest?
> 
> If you suggest numbered headlines, then I have an answer right away:
> This article isn't cookbook, it is a list of useful tips that might help. Not steps implementator must do. That's why there aren't numbers in headlines.

I am so sorry that I forgot it during first review and I did not prioritize this, but main reason for this article is to say to people: "Hey guys, you need to do this five steps because they are important"

Your style is "how to do something" and it should be "do this and this".

File: docs/introduction/start-building-your-application.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (simara-svatopluk) [reply docs/introduction/start-building-your-application.md]: I'm building a new project that will run on `a.com` and `b.com`, when I want to release this project to production, I have to change at least URLs in production.

And that is a different scenario than "already running application"

File: docs/introduction/start-building-your-application.md [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+
+We highly recommend to set up domains first. If you set up domains in the beginning of the project correctly, you'll save a lot of time.
+
+*Note: When you change the domain settings, it is likely that [tests](/docs/introduction/automated-testing.md) (mainly [acceptance](/docs/introduction/automated-testing.md#acceptance-tests-aka-functional-tests-or-selenium-tests)) will fail and that [demo data](/docs/introduction/basic-and-demo-data-during-application-installation.md) will not fit you.
+It is up to you to adjust them accordingly for your needs.*
+
+*Note 2: If you add a domain, please create and upload an icon for the new domain. You'll make shop administrators happy.*
```
Comment (LukasHeinz) [inline docs/introduction/start-building-your-application.md]:
```suggestion
*Note 2: If you add a domain, please create and upload an icon for the new domain (Administration > Settings >  E-shop identification). You'll make shop administrators happy.*
```

File: docs/introduction/start-building-your-application.md [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+* `shopsys` *(1st domain)*: `en`
+* `2.shopsys` *(2nd domain)*: `cs`
+* administration: `en`
+
+In case you want to change domain locale read [locale settings](/docs/introduction/how-to-set-up-domains-and-locales.md#3-locale-settings) or in case you want to change default administration locale read [locale in administration](/docs/introduction/how-to-set-up-domains-and-locales.md#36-locale-in-administration).
+
+## Set up Elasticsearch
+
+We use Elasticsearch on the frontend for product searching, filtering and for fast listing of products to provide better performance.
+You are likely to adjust the Elasticsearch configuration if you have a technical shop where inflection of product names doesn't make sense (we use inflection during searching by default).
```
Comment (LukasHeinz) [inline docs/introduction/start-building-your-application.md]:
```suggestion
You are likely to adjust the Elasticsearch configuration for example if you have a technical shop where the inflection of product names doesn't make sense (we use inflection during searching by default).
```

**Pull request merged**